digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataExporter/AssetReportCsvExporter.scala

```diff
@@ -0,0 +1,264 @@
+                                              else (elem.totalAssets, "")
+
+                val operatorDate = extractDateToString(elem.operatorModifiedDate)
+                val municipalityDate = extractDateToString(elem.municipalityModifiedDate)
+
+                val baseRow = s";${elem.assetNameFI};$totalPoints;$isLinear;$operatorDate;$municipalityDate;${elem.operatorUser};${elem.municipalityUser}"
+
+                if (elem.assetTypeId == TrafficSigns.typeId) {
+                  val totalNewLawTrafficSigns = getTotalNewLawTrafficSigns(municipality)
+                  baseRow.concat(s";$totalNewLawTrafficSigns")
+                }
+                else
+                  baseRow
+                }
+
+          line + rows.mkString("\r\n")
+        }
+
+      result.toList
+    }
+
+    output.mkString("\r\n")
+  }
+
+
+  def decodeAssetsToProcess(assetTypesList: List[Int]): List[Int] = {
+
+    val ( pointAssets, linearAssets ) = ( assetTypesList.contains(PointAssets.id), assetTypesList.contains(LinearAssets.id) )
+
+    if ( assetTypesList.contains(AllAssets.id) ) {
+      AssetTypeInfo.values.map(_.typeId).toList
+    }
+    else if ( pointAssets || linearAssets ) {
+
+      val (linearAssetType, pointAssetType) = AssetTypeInfo.values.partition( _.geometryType == "linear" )
+
+      val allPoints = pointAssetType.map(_.typeId)
+      val allLinears = linearAssetType.map(_.typeId)
+
+      (pointAssets, linearAssets) match {
+        case (true, false) =>
+          (allPoints ++ assetTypesList).filterNot(_ == PointAssets.id).toList
+
+        case (false, true) =>
+          (allLinears ++ assetTypesList).filterNot(_ == LinearAssets.id).toList
+
+        case (true, true) =>
+          (allPoints ++ allLinears).filterNot(id => id == PointAssets.id || id == LinearAssets.id).toList
+
+        case _ =>
+          List()
+      }
+
+    }
+    else
+      assetTypesList
+  }
+
+
+  def decodeMunicipalitiesToProcess(municipalities: List[Int]): List[Int] = {
+    if (municipalities.isEmpty || municipalities.contains(1000)) { /* 1000 = All municipalities */
+      withDynSession {
+        municipalityDao.getMunicipalitiesInfo.map(_._1).toList
+      }
+    }
+    else
+      municipalities
+  }
+
+
+  def exportAssetsByMunicipalityCSVGenerator(assetTypesList: List[Int], municipalitiesList: List[Int], logId: Long, withTransaction: Boolean = true): Long = {
+
+    try {
+
+      val finalAssetList = decodeAssetsToProcess(assetTypesList)
+      val (linearAssets, pointAssets) = finalAssetList.partition( AssetTypeInfo(_).geometryType == "linear")
+
+      val linearContent = if (linearAssets.nonEmpty) exportAssetsByMunicipality(assetTypesList, municipalitiesList, withTransaction = withTransaction)
+                        else List()
+
+      val pointContent = if (pointAssets.nonEmpty) exportAssetsByMunicipality(assetTypesList, municipalitiesList, extractPointsAsset = true, withTransaction)
+                        else List()
```

Comment:
Pattern matching could be what you are looking here

val (linearContent,  pointContent) = (linearAssets.nonEmpty, pointAssets.nonEmpty) match ...

or even val allContent = (same match above) since you are not looking to do anything with the linear/point content

Reply:
Just a reminder that we shouldn't use pattern matching to only check for true or false because it is heavier than an if else

Reply:
> Pattern matching could be what you are looking here
> 
> val (linearContent, pointContent) = (linearAssets.nonEmpty, pointAssets.nonEmpty) match ...
> 
> or even val allContent = (same match above) since you are not looking to do anything with the linear/point content

Don't know the vantage on that.
Because I will create 4 cases inside a match when 2 IF...else do the job in a simple way.

Reply:
Yeah you are right I was thinking too much Scala-like in this one.